msprime/mutations.py

```diff
@@ -131,9 +153,42 @@ def __init__(self):
 
 class HKY(MatrixMutationModel):
     """
-    The Hasegawa, Kishino and Yano mutation model (Hasegawa et al. 1985).
-
-    .. todo: documentation
+    The Hasegawa, Kishino and Yano 1985 mutation model, that allows an arbitrary
+    equilbrium base distribution, and has different transition/transversion rates.
+
+    Like other MatrixMutationModels, this is parameterized in terms of an overall
+    mutation rate (mu) and a transition probability matrix (P), so the rate at which
+    an instance of the ``i``th allele mutates to the ``j``th allele is
+    :math:``1/(mu * P[i,j])`` generations. In this model, :math:``\\kappa`` is
+    the transition:transversion ratio and :math:``\\pi`` is the vector of equilbrium
+    frequencies, so that we define :math:``P`` by first defining
+    :math:``p[i,j] = \\pi[j]`` if :math:``i \to j`` is a transversion, and
+    ``p[i,j] = \\kappa \\pi[j]`` otherwise, and then letting
+    :math:``P[i,j] = p[i,j] / M`` where :math:``M = 1 + (\\gamma - 1) \\max(\\pi)``
+    is the largest row sum of :math:``p``. This means that if you mutate with the
+    HKY model with ``mutation_rate`` equal to :math:``\\mu``, then the mutation
+    rate from "A" to "G", for instance, is
+    :math:``\\mu \\kappa \\pi[G] / (1 + (\\gamma - 1) \\max(\\pi))``.
```

Comment:
This lengthy explanation of how to get the actual mutation rates from the parameters is kinda unfortunate. You must have come across this, @GertjanBisschop - what do you think? We could change the mutation generation code to avoid this, by not requiring the the transition probability matrix have row sums equal to one (and doing the renormalization at the C level), or adding an extra "rate multiplier" property to mutation models to make this `M` go away? Or, probably, we should just say it's a minor annoyance and leave well enough alone?

Reply:
A few more thoughts: I was looking at [the HKY model on Wikipedia](https://en.wikipedia.org/wiki/Models_of_DNA_evolution#HKY85_model_(Hasegawa,_Kishino_and_Yano_1985)), and thinking that their rate matrix is much nicer: e.g. Q[A, C] = pi[A] instead of mu * pi[A] / (1 + (gamma-1) * max(pi)). But, Wikipedia is *not* giving the rate matrix, just giving it up to an overall constant.  So, Wikpedia's parameterization is Q[A,C] = mu * pi[A], which is simpler than ours, but we just have a somewhat different interpretation of mu - their mu is our mu / M. So, I think maybe our parameterization is fine?

Reply:
I moved this to docs/api.rst, so that I could get it to compile, and rewrote it, btw.

Reply:
I'm trying to figure out if our parameterization *does* differ from seq-gen's? I see that kappa differs by a factor of 2, which I understand, but how about the overall mutation rate? Is this something you got to the bottom of, @GertjanBisschop? Gah, I wish seq-gen documented what it's doing.